src/nemos/base_class.py

```diff
@@ -180,6 +238,181 @@ class BaseRegressor(Base, abc.ABC):
+        ------
+        NameError
+            If any of the solver keyword arguments are not valid.
+        """
+        solver_args = inspect.getfullargspec(getattr(jaxopt, solver_name)).args
+        undefined_kwargs = set(solver_kwargs.keys()).difference(solver_args)
+        if undefined_kwargs:
+            raise NameError(
+                f"kwargs {undefined_kwargs} in solver_kwargs not a kwarg for jaxopt.{solver_name}!"
+            )
+
+    def instantiate_solver(
+        self,
+        loss: Callable,
+        *args: Any,
+        prox: Optional[Callable] = None,
+        **kwargs: Any
+    ) -> Tuple[SolverInit, SolverUpdate, SolverRun]:
+        """
+        Instantiate the solver with the provided loss function.
+
+        Instantiate the solver with the provided loss function, and return callable functions
+        that initialize the solver state, update the model parameters, and run the optimization.
+
+        This method creates a solver instance from jaxopt library, tailored to the specific loss
+        function and regularization approach defined by the Regularizer instance. It also handles
+        the proximal operator if required for the optimization method. The returned functions are
+         directly usable in optimization loops, simplifying the syntax by pre-setting
+        common arguments like regularization strength and other hyperparameters.
+
+        Parameters
+        ----------
+        loss :
+            The loss function to be optimized.
+
+        *args:
+            Positional arguments for the jaxopt `solver.run` method, e.g. the regularizing
+            strength for proximal gradient methods.
+
+        prox:
+            Optional, the proximal projection operator.
+
+        *kwargs:
+            Keyword arguments for the jaxopt `solver.run` method.
+
+        Returns
+        -------
+        :
+            A tuple containing three callable functions:
+            - solver_init_state: Function to initialize the solver's state, necessary before starting the optimization.
+            - solver_update: Function to perform a single update step in the optimization process,
+            returning new parameters and state.
+            - solver_run: Function to execute the optimization process, applying multiple updates until a
+            stopping criterion is met.
+        """
+        # check that the loss is Callable
+        utils.assert_is_callable(loss, "loss")
+
+        # final check that solver is valid for chosen regularizer
+        if self.solver_name not in self.regularizer.allowed_solvers:
+            raise ValueError(f"The solver: {self.solver_name} is not allowed for "
+                             f"{self._regularizer.__class__} regularizaration. Allowed solvers are "
+                             f"{self._regularizer.allowed_solvers}.")
+
+        # get the solver with given arguments.
+        # The "fun" argument is not always the first one, but it is always KEYWORD
```

Comment:
Great, thanks for going through the jaxopt API. I feel like they are not too good with consistency at times. Did you have to look at all their solvers or is there some documentation that explains how the solver arguments are structured? If that document exist, i would add a link to it here in the comments